docs/spec/lcd/api.md

```diff
@@ -0,0 +1,847 @@
+    },
+    "rest api": "2.0"
+}
+```
+
+* The above command returns JSON structured like this if fails:
+
+```
+{
+    "error": "Insufficient Funds",
+    "code":500,
+    "result": {},
+    "rest api": "2.0"
+}
+```
+
+
+### /signed_transfer - POST
+
+url: /signed_transfer, Method: POST
+
+Functionality: transfer asset
+
+Parameters:
+
+| Parameter       | Type   | Default | Required | Description            |
+| ------------    | ------ | ------- | -------- | ----------------------------------------------- |
+| signed_transfer | []byte | null    | true     | bytes of a valid transaction and it's signature |
+
+* The above command returns JSON structured like this if success:
+
+```
+{
+    "error": "",
+    "code":200,
+    "result": {
+     "tx_hash": ""
+    },
+    "rest api": "2.0"
+}
+```
+
+* The above command returns JSON structured like this if fails:
+
+```
+{
+    "error": "Invalid Signature",
+    "code":500,
+    "result": {},
+    "rest api": "2.0"
+}
+```
+
+
+## ICS21 - StakingAPI
+
+The StakingAPI exposes all functionality needed to stake tokens. It is exposed by the staking
+module.
+
+
+### /stake/{delegator}/bonding_status/{validator} - GET
+
+url: /stake/{delegator}/bonding_status/{validator}, Method: GET
+
+Functionality: get delegator information and verify returned proof
+
+Parameters: null
+
+* The above command returns JSON structured like this if success:
+
+```
+{
+    "error": "",
+    "code":200,
+    "result": {
+     "validator":address1
+     "delegator":address2
+     "liability": 1
+    },
+    "rest api": "2.0"
+}
+```
+
+* The above command returns JSON structured like this if fails:
+
+```
+{
+    "error": "cannot send asset to the account",
+    "code":500,
+    "result": {},
+    "rest api": "2.0"
+}
+```
+
+
+### /stake/validators - GET
+
+url: /stake/validators, Method: GET
+
+Functionality: get all validators information in stake module and verify returned proof
+
+Parameters: null
+
+* The above command returns JSON structured like this if success:
+
+```
+{
+    "error": "",
+    "code":200,
+    "result": {
+      "validatorset": [
+          {
+            "name": "monkey",
+            "address": "cosmosaccaddr1fedh326uxqlxs8ph9ej7cf854gz7fd5zlym5pd",
+            "pub_key": "cosmosaccpub1zcjduc3q8s8ha96ry4xc5xvjp9tr9w9p0e5lk5y0rpjs5epsfxs4wmf72x3shvus0t",
+            power:1000
+          },
+   		 {
+            "name": "test",
+            "address": "cosmosaccaddr1thlqhjqw78zvcy0ua4ldj9gnazqzavyw4eske2",
+            "pub_key": "cosmosaccpub1zcjduc3qyx6hlf825jcnj39adpkaxjer95q7yvy25yhfj3dmqy2ctev0rxmse9cuak",
+             power:1000
+         }
+	],
+    "block_height": 5241
+    }   
+    },
+    "rest api": "2.0"
+}
+```
+
+* The above command returns JSON structured like this if fails:
+
+```
+{
+    "error": "cannot send asset to the account",
+    "code":500,
+    "result": {},
+    "rest api": "2.0"
+}
+```
+
+
+### /stake/delegations - POST
```

Comment:
How about a more REST-ful approach: PUT `/stake/candidates/{candidate}/delegator/{delegator/me}` {amount, name, password}
This serves for bonding and updating the bond.